test/new-e2e/tests/process/k8s_test.go

```diff
@@ -73,12 +79,125 @@ func TestK8sTestSuite(t *testing.T) {
+		"The bug appears to be in test-infra-definitions and it's default helm values taking precedence")
+	t := s.T()
+	helmValues, err := createHelmValues(helmConfig{
+		ProcessAgentEnabled:        true,
+		ProcessDiscoveryCollection: true,
+	})
+	require.NoError(t, err)
+
+	s.UpdateEnv(awskubernetes.KindProvisioner(
+		awskubernetes.WithWorkloadApp(func(e config.Env, kubeProvider *kubernetes.Provider) (*kubeComp.Workload, error) {
+			return cpustress.K8sAppDefinition(e, kubeProvider, "workload-stress")
+		}),
+		awskubernetes.WithAgentOptions(kubernetesagentparams.WithHelmValues(helmValues)),
+	))
+
+	assert.EventuallyWithT(t, func(collect *assert.CollectT) {
+		status := s.getAgentStatus()
+		assert.ElementsMatch(t, []string{"process_discovery"}, status.ProcessAgentStatus.Expvars.Map.EnabledChecks)
+	}, 2*time.Minute, 5*time.Second)
+
+	var payloads []*aggregator.ProcessDiscoveryPayload
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		var err error
+		payloads, err = s.Env().FakeIntake.Client().GetProcessDiscoveries()
+		assert.NoError(c, err, "failed to get process discovery payloads from fakeintake")
+		assert.NotEmpty(c, payloads, "no process discovery payloads returned")
+	}, 2*time.Minute, 10*time.Second)
+
+	assertProcessDiscoveryCollected(t, payloads, "stress-ng-cpu [run]")
+}
+
+func (s *K8sSuite) TestManualProcessCheck() {
+	agent := getAgentPod(s.T(), s.Env().KubernetesCluster.Client())
+
+	// The log level needs to be overridden as the pod has an ENV var set.
+	// This is so we get just json back from the check
+	stdout, stderr, err := s.Env().KubernetesCluster.KubernetesClient.
+		PodExec(agent.Namespace, agent.Name, "process-agent",
+			[]string{"bash", "-c", "DD_LOG_LEVEL=OFF process-agent check process -w 5s --json"})
+	assert.NoError(s.T(), err)
+	assert.Empty(s.T(), stderr)
+
+	assertManualProcessCheck(s.T(), stdout, false, "stress-ng-cpu [run]", "stress-ng")
+}
+
+func (s *K8sSuite) TestManualProcessDiscoveryCheck() {
+	agent := getAgentPod(s.T(), s.Env().KubernetesCluster.Client())
+	// The log level needs to be overridden as the pod has an ENV var set.
+	// This is so we get just json back from the check
+	stdout, stderr, err := s.Env().KubernetesCluster.KubernetesClient.
+		PodExec(agent.Namespace, agent.Name, "process-agent",
+			[]string{"bash", "-c", "DD_LOG_LEVEL=OFF process-agent check process_discovery -w 5s --json"})
+	assert.NoError(s.T(), err)
+	assert.Empty(s.T(), stderr)
+
+	assertManualProcessDiscoveryCheck(s.T(), stdout, "stress-ng-cpu [run]")
+}
+
+func (s *K8sSuite) TestManualContainerCheck() {
+	agent := getAgentPod(s.T(), s.Env().KubernetesCluster.Client())
+
+	// The log level needs to be overridden as the pod has an ENV var set.
+	// This is so we get just json back from the check
+	stdout, stderr, err := s.Env().KubernetesCluster.KubernetesClient.
+		PodExec(agent.Namespace, agent.Name, "process-agent",
+			[]string{"bash", "-c", "DD_LOG_LEVEL=OFF process-agent check container -w 5s --json"})
```

Comment:
Should we maybe increase this to 10 seconds to help avoid potential flakiness?

Reply:
I'm going to leave it as 5s for now as the number its an arbitrary delay to wait for workloadmeta. If it starts to flake we can increase it.